src/predicate/isSymbol.ts

```diff
@@ -0,0 +1,15 @@
+import { getTag } from '../compat/_internal/getTag';
+
+/**
+ * Check whether a value is a symbol.
+ *
+ * This function can also serve as a type predicate in TypeScript, narrowing the type of the argument to `symbol`.
+ *
+ * @param {unknown} value The value to check.
+ * @returns {value is symbol} Returns `true` if `value` is a symbol, else `false`.
+ */
+export function isSymbol(value: unknown): value is symbol {
+  return (
+    typeof value === 'symbol' || (value !== null && typeof value === 'object' && getTag(value) === '[object Symbol]')
+  );
```

Comment:
While this is great, I think supporting `Object(symbol)`s is not desirable in our library (since I think it is an edge case, and it makes our implementation complicated).

Why not add this branch in our compatibility library, `es-toolkit/compat`?

For example, we have two versions of `isBoolean`:

- [one](https://github.com/toss/es-toolkit/blob/main/src/predicate/isBoolean.ts) in our original library
- [another](https://github.com/toss/es-toolkit/blob/main/src/compat/predicate/isBoolean.ts) in our compatibility library.